### PR TITLE
test: add plugin e2e tests

### LINKED ISSUE #387: [REQ] add tests for plugins
### What kind of request is this?

Improvement of existing experience

### What is your request or suggestion?

add test for scanner plugin flow

### Are you willing to submit PRs to contribute to this feature request?

- [ ] Yes, I am willing to implement it.

### PR DESCRIPTION
**Pull request by anubhav06**
Add e2e tests for plugin

Closes #387

### CODE REVIEW

**Comment by anubhav06** [comment]
ready for review, is this fine?

**Comment by sozercan** [comment]
@anubhav06 sorry didn't get a chance to look at this yet. I will review after kubecon.

File: test/e2e/plugin_test.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+		{
+			image:  "docker.io/library/nginx:1.23",
+			report: "./testdata/fake_report.json",
+			err:    fmt.Errorf("Error: unsupported osType FakeOS specified\n"), // nolint:revive
+		},
+		{
+			image:  "docker.io/library/alpine:3.14.0",
+			report: "./testdata/invalid_report.json",
+			err:    fmt.Errorf("Error: error running scanner fake: exit status 1\n"), // nolint:revive
+		},
```
Comment (sozercan) [inline test/e2e/plugin_test.go:26]: can we add a valid case too? we can add a super simple `valid_report.json` 

for example, something like:
```
{
    "OSType": "alpine",
    "OSVersion": "3.7.3",
    "Arch": "amd64",
    "Packages": [
        {
            "Name": "musl",
            "InstalledVersion": "1.1.18-r3",
            "FixedVersion": "1.1.18-r4",
            "VulnerabilityID": "CVE-2019-14697"
        }
    ]
}
```

File: test/e2e/plugin_test.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.image, func(t *testing.T) {
```
Comment (sozercan) [inline test/e2e/plugin_test.go:30]: can we add parallel?

File: test/e2e/testdata/fake_report.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+            "VulnerabilityID": "VULN002"
+        },
+        {
+            "Name": "baz",
+            "InstalledVersion": "3.0.0",
+            "FixedVersion": "",
+            "VulnerabilityID": "VULN003"
+        }
+    ]
+}
```
Comment (sozercan) [inline test/e2e/testdata/fake_report.json:25]: nit: newline

File: test/e2e/testdata/invalid_report.json [hunk 4]
```diff
@@ -0,0 +1 @@
+invalid
```
Comment (sozercan) [inline test/e2e/testdata/invalid_report.json:1]: nit: newline

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -128,3 +128,37 @@ jobs:
           set -eu -o pipefail
           . .github/workflows/scripts/buildkitenvs/${{ matrix.buildkit_mode}}
           go test -v ./integration --addr="${COPA_BUILDKIT_ADDR}" --copa="$(pwd)/copa" -timeout 0
+
+  test-plugin:
+    needs: test-patch
```
Comment (sozercan) [inline .github/workflows/build.yml:133]: why do we need to wait for `test-patch`? would `needs: build` work?

**Review by sozercan — CHANGES_REQUESTED** [review]
@anubhav06 sorry about the delay! added a few minor comments

**Comment by anubhav06** [comment]
I've my university exams going on so I'll fix this after 3-4 days. Thanks :)

**Comment by sozercan** [comment]
good luck on the exam!

File: .github/workflows/build.yml (hunk 5 above)
Comment (anubhav06) [reply .github/workflows/build.yml]: yes, added

File: test/e2e/plugin_test.go [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+func TestPlugins(t *testing.T) {
+	testCases := []struct {
+		image  string
+		report string
+		err    error
+	}{
+		{
+			image:  "docker.io/library/alpine:3.14.0",
+			report: "./testdata/invalid_report.json",
+			err:    fmt.Errorf("exit status 1"), // nolint:revive
```
Comment (sozercan) [inline test/e2e/plugin_test.go:20]: why nolint here and below?

**Review by sozercan — APPROVED** [review]
small nit. otherwise, great work! LGTM

**Review by sozercan — APPROVED** [review]
Thank you! LGTM

**Pull request merged**